app/authors/page.js

```diff
@@ -0,0 +1,143 @@
+import Author from "@/models/authors";
+import Book from "@/models/books";
+import Link from "next/link";
+import { RxAvatar } from "react-icons/rx";
+import { Op } from "sequelize";
+import AuthorItem from "../components/AuthorItem";
+import BookItem from "../components/BookItem";
+import EntriesPerPageDropdown from "../components/entriesPerPage";
+import Pagination from "../components/Pagination";
+import SearchForm from "../components/SearchForm";
+import { validateImageUrl } from "../utils/imageValidator";
+import { getPaginationParams } from "../utils/pagination";
+import CreateAuthorForm from "./CreateAuthorForm";
+
+export default async function ServerAuthorsComponent({ params, searchParams }) {
+  const { page, size, searchTerm } = getPaginationParams(searchParams);
+  let data = [];
+  const offset = page * size;
+  const where = searchTerm
+    ? {
+        name: {
+          [Op.iLike]: `%${searchTerm}%`,
+        },
+      }
+    : {};
+
+  const { rows, count } = await Author.findAndCountAll({
+    where,
+    limit: size,
+    offset: offset,
+    include: [Book],
+    distinct: true,
+  });
+
+  const validatedAuthors = await Promise.all(
+    rows.map(async (author) => ({
+      ...author.toJSON(),
+      profilePhotoUrl: await validateImageUrl(author.profilePhotoUrl, "author"),
+      Books: await Promise.all(
+        author.Books.map(async (book) => ({
+          ...book.toJSON(),
+          profilePhotoUrl: await validateImageUrl(book.profilePhotoUrl, "book"),
+        }))
+      ),
+    }))
+  );
```

Comment:
this logic seems duplicated, its present in api/route as well, what can we do here?